mavlink_vehicles/mavlink_vehicles.cc

```diff
@@ -129,12 +212,31 @@ void msghandler::handle(mav_vehicle &mav, const mavlink_message_t *msg)
     case MAVLINK_MSG_ID_HEARTBEAT: {
         mavlink_heartbeat_t hb;
         mavlink_msg_heartbeat_decode(msg, &hb);
-        mav.base_mode = (hb.base_mode & MAV_MODE_FLAG_GUIDED_ENABLED)
-                            ? mode::GUIDED
-                            : mode::OTHER;
+
+        // Decode mode flag ArduCopter only supports custom_modes. The
+        // custom_mode value for each mode is described on ArduCopter's
+        // documentation.
+        mav.base_mode = mode::OTHER;
+
+        if (!(hb.base_mode & MAV_MODE_FLAG_CUSTOM_MODE_ENABLED)) {
+            break;
+        }
+
+        switch (hb.custom_mode) {
+        case 3:
```

Comment:
It's a custom ardupilot mode as defined in [here](https://github.com/ArduPilot/ardupilot/blob/master/ArduCopter/defines.h#L91). I've added one more comment to be more explicit.